DM-5940: Create new build based on the converted firefly code.

loitly opened the pull request:
- separate firefly viewer from fftools source directory
  - fftools is the deprecated GWT version
  - firefly viewer is the new javascript version with minimal gwt code.
- create scheme to load fresh javascript using static name.
  - generate *_loader.js file to dynamically load the most recent built js file.
  - set *_loader.js to no-cache.
- add version info to drop down panel.
  - no longer rely on rpc call.

minor:
- cleanup webpack build script
- fix old fftools’s catalog search bug
- fix table select all bug

For additional information:  https://jira.lsstcorp.org/browse/DM-5940

tgoldina commented on buildScript/webpack.config.js:
```diff
@@ -130,102 +221,47 @@ var webpackConfig = {
-    	];
-	}
-}
+            console.log('!!!!!!!!' + path.resolve(outpath, `${stats.compilation.name}.nocache.js`));
```
Should it be here? Build script outputs
...
!!!!!!!!/hydra/cm/firefly/build/firefly/gwt/firefly/firefly.nocache.js
good!!!!!!
after good!!!!!!
...
Looks like leftovers from debugging.

tgoldina commented on src/firefly/js/ui/DropDownContainer.css:
```diff
@@ -28,6 +28,12 @@
 }
 
+.DD-ToolBar__version {
+    color: rgba(256,256,256,0.1);
```
Currently, the version is virtually impossible to read, even when selected. I am wondering if it's possible to make it more readable with a lighter color, at least on mouse over.

tgoldina commented on src/firefly/js/util/WebUtil.js:
```diff
@@ -351,18 +354,6 @@ export function updateSet(object, path, value) {
 }
 
 /**
- * This is a wrapper of React update's $merge for use with deep object update.
```
Why are you removing updateMerge?

loitly replied:
I removed it thinking no one was using it.  Then I realized quickly that Emmanuel was using it.  I will add it back in.

ejoliet replied:
yes, please. Useful! Thanks for bringing back.

On 5/9/16 1:29 PM, loitly wrote:
>
> In src/firefly/js/util/WebUtil.js 
> <https://github.com/Caltech-IPAC/firefly/pull/80#discussion_r62567320>:
>
> > @@ -351,18 +354,6 @@ export function updateSet(object, path, value) {
> >  }
> >
> >  /**
> > - * This is a wrapper of React update's $merge for use with deep object update.
>
> I removed it thinking no one was using it. Then I realized quickly 
> that Emmanuel was using it. I will add it back in.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/Caltech-IPAC/firefly/pull/80/files/8f4a66e088ebb41291604de6708df1d8dcabd878#r62567320>
>

-- 
Emmanuel Joliet
NASA/IPAC  Infrared Science Archive
MS 100-22, Caltech, Pasadena, CA 91125
Office: 170
Email:  ejoliet@ipac.caltech.edu
Phone:  (626) 395-1489

tgoldina commented:
The code looks good to me. I built fftools and firefly, and both http://localhost:8080/firefly and http://localhost:8080/fftools work.

- http://localhost:8080/fftools/fftools.html shows loading icon

Several other things I noticed are now broken - probably in earlier tickets:
- when I move mouse over the plot buttons  in Firefox, the split panel reloads
- layout buttons (like tbl-xy, img-tbl) do not work any more 
- highlighted row is not set correctly when clicking on xyplot (with multipage catalog)

loitly commented:
- removed console log
- fixed server not caching query when it should
- fixed highlighted row not set correctly when clicking on xyplot with multiple pages
- fixed layout switch buttons not working anymore
- made version info more readable.
- add back updateMerge
